src/compiler/typescript_to_rust/azle_generate/src/candid_type/type_ref/mod.rs

```diff
@@ -69,15 +69,25 @@ impl SourceMapped<'_, TsTypeRef> {
         if let Some(primitive) = self.to_primitive()? {
             return Ok(CandidType::Primitive(primitive));
         }
-        Ok(match self.get_name()? {
-            "Opt" => CandidType::Opt(self.to_option()?),
-            "Func" => CandidType::Func(self.to_func(None)?),
-            "Record" => CandidType::Record(self.to_record()?),
-            "Tuple" => CandidType::Tuple(self.to_tuple()?),
-            "Vec" => CandidType::Array(self.to_vec()?),
-            "Variant" => CandidType::Variant(self.to_variant()?),
-            _ => CandidType::TypeRef(self.to_type_ref()?),
-        })
+        if let Some(opt) = self.to_option()? {
```

Comment:
Let's bring the match back

Reply:
We we are moving to Result<Option> then the if let some is going to be the most concise way to handle this.